fiftyone/core/stages.py

```diff
@@ -5205,26 +5220,122 @@ class SelectFields(ViewStage):
             May contain ``embedded.field.name`` as well
     """
 
-    def __init__(self, field_names=None, _allow_missing=False):
+    def __init__(
+        self, field_names=None, _allow_missing=False, _meta_filter=None
+    ):
         if etau.is_str(field_names):
             field_names = [field_names]
         elif field_names is not None:
             field_names = list(field_names)
 
         self._field_names = field_names
         self._allow_missing = _allow_missing
+        self._meta_filter = _meta_filter
 
     @property
     def field_names(self):
         """The list of field names to select."""
         return self._field_names or []
 
+    @property
+    def meta_filter(self):
+        """The meta filters to apply, as a string or dict."""
+        return self._meta_filter or {}
+
     def get_selected_fields(self, sample_collection, frames=False):
         if frames:
             return self._get_selected_frame_fields(sample_collection)
 
         return self._get_selected_fields(sample_collection)
 
+    def get_meta_filtered_fields(self, schema):
```

Comment:
Are you open to converting this to a private method and then simply calling it from inside the existing `get_selected_fields()` method?

From the perspective of users, I don't think it matters whether this stage is selecting fields based on an explicit list of `field_names` or a dynamic set of field names inferred based on a provided `meta_filter`.

With this change, we wouldn't need to add the new `get_meta_filtered_fields()` method to the `ViewStage` interface.

For context, the only reason `get_selected_fields()` and `get_excluded_fields()` were added to the interface is that it is important for the rest of the library to know if a view omits fields from the full dataset, as this has implications on what kind of database operations can be used when saving edits to samples. But it's not important whether the selected fields were chosen in a fancy way.